src/composables/useNavigationGuards.ts

```diff
@@ -1,91 +1,24 @@
 import NProgress from 'nprogress';
-import { Router, useRouter } from 'vue-router';
-
-import { Network } from '@balancer-labs/sdk';
-
+import { useRouter } from 'vue-router';
+import { networkId } from './useNetwork';
 import { useSidebar } from './useSidebar';
 import useVeBal from './useVeBAL';
-import {
-  networkFromSlug,
-  getSubdomain,
-  handleNetworkUrl,
-  networkId,
-  networkSlug,
-  appUrl,
-} from '@/composables/useNetwork';
 
 // Progress bar config
 NProgress.configure({ showSpinner: false });
 let delayedStartProgressBar;
 
-function setNetworkAndRefresh(networkId: string, url: string, router: Router) {
-  document.write('');
-  localStorage.setItem('networkId', networkId);
-  window.location.href = url;
-  router.go(0);
-}
-
+/**
+ * Navigation gaurds that require Vue app context.
+ */
 export default function useNavigationGuards() {
   const router = useRouter();
   const { setShowRedirectModal, isVeBalSupported } = useVeBal();
   const { setSidebarOpen } = useSidebar();
 
-  /**
-   * Subdomain redirects - e.g. "polygon.balancer.fi/".
-   */
   router.beforeEach((to, from, next) => {
-    const subdomain = getSubdomain(window.location.host);
-    const subdomainNetwork = networkFromSlug(subdomain);
-    if (subdomainNetwork) {
-      setNetworkAndRefresh(
-        subdomainNetwork.toString(),
-        `${appUrl().replace(subdomain, 'app')}${
-          to.params.networkSlug ? '' : networkSlug
-        }${to.fullPath}`,
-        router
-      );
-    }
-    next();
-  });
+    localStorage.setItem('networkId', networkId.value.toString());
```

Comment:
I've moved networkId localstorage setting here because it didn't feel right to have to do it in multiple places in nav-gaurds.ts. It also made some of those functions more awkward to write which made me think it was the wrong place to be setting it.

Reply:
I also think it makes more sense to sync the localstorage with the app's current state in the app context based nav guards, which is what this file is for. I'm not 100% sure if this will break expected behaviour or not. Let me know if you think it will.